models/marts/dim_customers.sql

```diff
@@ -0,0 +1,36 @@
+{{ config(materialized='table') }}
+
+with customers as (
+    select * from {{ ref('stg_customers') }}
+),
+orders as (
+    select * from {{ ref('stg_orders') }}
+),
+payments as (
+		select * from {{ ref('stg_payments') }}
+),
+customer_order_aggregates as (
+    select
+        orders.customer_id,
+        min(orders.order_date) as first_order_date,
+        max(orders.order_date) as most_recent_order_date,
+        count(orders.order_id) as number_of_orders,
```

Comment:
This will actually not sum to the number of `orders` but instead it will sum to the number of `payments`. This is because each order can have more than 1 payment, so when you left joined `payments` it actually fanned out the table. The count you will get here is the total count of payments each customer has made. You can solve this by having a cte that summarizes payments before this!